From 727a1d122c904ceec48cacdf5ea8e729a90e456b Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 9 Sep 2024 14:36:42 +0200
Subject: [PATCH 1077/1695] dts: stm32: describe vrefbuf supply for stm32mp21

Describe PWR VDDA18ADC as VREFBUF supply for STM32MP21.
The goal is to guaranty that VDDA18ADC SV is set before enabling VREFBUF.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 603e816a8af3f1bc94d7d35b6f850918469c583c dts: stm32: describe vrefbuf supply for stm32mp21]
Change-Id: I6e611cf17f0b219706f4506a66776612a8b07b29
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/403073
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index d5da6cbc5..33efc19fb 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -634,6 +634,7 @@
 			regulator-min-microvolt = <1210000>;
 			regulator-max-microvolt = <1500000>;
 			clocks = <&rcc CK_BUS_VREF>;
+			vdda-supply = <&vdda18adc>;
 			status = "disabled";
 		};
 
-- 
2.47.2

